File: DSAlgorithms/Leetcode/c++/70-climbingStairs.cpp
#include <iostream>

int climbingStairs(int n)
{
    int one = 1;
    int two = 1;
    int temp;
    while(n > 1)
    {
        temp = one;
        one = one + two;
        two = temp;
        --n;
    }
    return one;
}


int climbingStairs2(int n)
{
    int one = 1;
    int two = 1;
    int temp;
    for(int i = 0; i < n - 1; i++)
    {
        temp = one;
        one = one + two;
        two = temp;
    }
    return one;
}

File: DSAlgorithms/Leetcode/ts/70-climbingStairs.ts
const climbingStairs = (n:number): number => {
    let one = 1;
    let two = 1;
    let temp;
    while(n > 1)
    {
        temp = one;
        one = one + two;
        two = temp;
        --n;
    }
    return one;
}

const climbingStairs2 = (n: number):number => {
    let one = 1; 
    let two = 1;
    let temp;
    for(let i = 0; i < n - 1; i++)
    {
        temp = one;
        one = one + two;
        two = temp;
    }
    return one;
}